### DISCUSSION TITLE
Why is [FAIL. Reason: MerkleAirdrop__InvalidProof()] happening when running the test in Section 4: Airdrop and Signatures?

### DISCUSSION THREAD

**Discussion by palmcivet7** [Cyfrin/foundry-full-course-cu#2081] [Q&A] [answered]
Every time I run the [test in Section 4: Airdrop and Signatures](https://github.com/Cyfrin/foundry-merkle-airdrop-cu/blob/b4c627c3bcce7ecb5bb3df1f5209dda94059626b/test/unit/MerkleAirdrop.t.sol#L45-L58), I've been getting this error:
```
├─ [8220] MerkleAirdrop::claim(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], 25000000000000000000 [2.5e19], [0x0fd7c981d39bece61f7499702bf59b3114a90e66b51ba2c53abdf7b62986c00a, 0xe5ebd1e1b5a5478a944ecab36a9a954ac3b6b8216875f6524caa7a1d87096576], 27, 0x6ef35186d5e0e854196ab143613e564caa45707c3c02c97f9583dc2525dea233, 0x2a1438d27acf1ae24e1bf18b44fd7c0f72025a3cf1cc7a44f1992bf615cd887c)
    │   ├─ [3000] PRECOMPILES::ecrecover(0x53018c2068ff89e09094bfc005a086c0881506460320b44131c0b996fa8ed110, 27, 50184319857511433265145613098790863537787021025560114461553746468679922393651, 19032868754110274016812456918942158897741509918834712645675547399228549335164) [staticcall]
    │   │   └─ ← [Return] 0x0000000000000000000000006ca6d1e2d5347bfab1d91e883f1915560e09129d
    │   └─ ← [Revert] MerkleAirdrop__InvalidProof()
```
I tried cloning the [lesson repo](https://github.com/Cyfrin/foundry-merkle-airdrop-cu) and got the same error when running the test there.

Why is that happening?

Edit: Just to be clear, the failure is happening when running `forge test` or `forge test --mt testUsersCanClaim` inside the lesson repo.

**Comment by EngrPips** [comment]
This happens if you are using a wrong proof or the right proof but passing in the wrong user. Check your code to see which is causing your revert. You can also share your test and script so we also help you look and maybe we can catch the issue.

**Reply by palmcivet7** [reply]
It's failing in this https://github.com/Cyfrin/foundry-merkle-airdrop-cu

**Reply by EngrPips** [reply]
My codebase works fine, I mean I don't clone the repo but I write my own code.

**Reply by ciaranightingale** [reply, marked as answer]
Hi @palmcivet7 you need to update the root in the deployment script: https://github.com/Cyfrin/foundry-merkle-airdrop-cu/blob/main/script/DeployMerkleAirdrop.s.sol
I have updated the repo, thanks for spotting this!
Double check the roots and proofs in the test (or interaction script) and root in the deployment script all match your `output.json` file:)
Hope this helps
